src/psnr.jl

```diff
@@ -0,0 +1,67 @@
+"""
+    PeakSignalNoiseRatio <: FullReferenceIQI
+    psnr(x, ref [, peakval])
+    assess(PSNR(), x, ref, [, peakval])
+
+Peak signal-to-noise ratio (PSNR) is used to measure the quality of image.
+
+PSNR (in dB) is calculated by `10log10(PeakVal^2/mse(x, ref))`, where `PeakVal`
+is the maximum possible pixel value of image `ref`, and
+[`mse`](@ref ImageDistances.mse) is the mean squared error between images `x`
+and `ref`.
+
+!!! info
+
+    For general `Color3` images, PSNR is reported against each channel of that
+    color space as a `Vector`. One exception is `AbstractRGB` images, where
+    `mse` is calculated over three channels by convention.
+
+!!! info
+
+    `peakval` is optional only for gray and RGB color space. For other color
+    spaces, you need to explicitly pass it as a `Vector`.
+"""
+struct PeakSignalNoiseRatio <: FullReferenceIQI end
+const PSNR = PeakSignalNoiseRatio # alias PSNR since it's too famous
+
+# api
+(iqi::PSNR)(x, ref, peakval) = psnr(x, ref, peakval)
+(iqi::PSNR)(x, ref) = iqi(x, ref, peak_value(eltype(ref)))
+
+@doc (@doc PeakSignalNoiseRatio)
+psnr(x, ref, peakval) = psnr(x, ref, peakval)
```

Comment:
It was `psnr(x, ref, peakval) = _psnr(x, ref, peakval)`, where `_psnr` is the real implementation.
I change it back to `psnr`, thought it would be more elegant with multiple dispatches. But with a second look at it, somehow I think it becomes slightly harder for people to figure out the dispatch order.

In case someone catches this naming bug, I'll fix it later

Reply:
This was a naming mistake (it should be `psnr(x, ref, peakval) = _psnr(x, ref, peakval)`), fixed by the new commit.